src/base/utils/string.cpp

```diff
@@ -144,30 +172,43 @@ namespace
 
 bool Utils::String::naturalCompareCaseSensitive(const QString &left, const QString &right)
 {
-    // provide a single `NaturalCompare` instance for easy use
-    // https://doc.qt.io/qt-5/threads-reentrancy.html
-#ifdef Q_OS_MAC  // workaround for Apple xcode: https://stackoverflow.com/a/29929949
-    static QThreadStorage<NaturalCompare> nCmp;
-    if (!nCmp.hasLocalData()) nCmp.setLocalData(NaturalCompare(true));
-    return (nCmp.localData())(left, right);
-#else
-    thread_local NaturalCompare nCmp(true);
-    return nCmp(left, right);
-#endif
+    return NaturalCompare::naturalCompareCaseSensitive(left, right) < 0;
```

Comment:
It wasn't to make the functions look nice, but to avoid:

1. Duplicating the code again;

2. Having extra thread-locals when we only need the 2.